lib/Unix/PID/Tiny.pm

```diff
@@ -1,12 +1,26 @@
 package Unix::PID::Tiny;
 
 use strict;
-$Unix::PID::Tiny::VERSION = 0.91;
+use warnings;
+
+our $VERSION = '0.92';
 
 sub new {
     my ( $self, $args_hr ) = @_;
+
+    my %DEFAULTS = (
+        'keep_open'           => 0,
+        'check_proc_open_fds' => 0,
+        'close_on_destroy'    => 0
+    );
+
+    $args_hr ||= {};
     $args_hr->{'minimum_pid'} = 11 if !exists $args_hr->{'minimum_pid'} || $args_hr->{'minimum_pid'} !~ m{\A\d+\z}ms;    # this does what one assumes m{^\d+$} would do
 
+    foreach my $key ( keys %DEFAULTS ) {
```

Comment:
just curious, why not: `%{$args_hr} = (%DEFAULTS, %{$args_hr});`

Reply:
You're right, they would be undef in the returned blessed object anyhow

Reply:
You're right :)

Reply:
However, one does read a lot more explicitly than the other, albeit the original form does cost a few extra ops.